test/module/shared_model/backend_proto/proto_query_response_factory_test.cpp

```diff
@@ -282,6 +283,47 @@ TEST_F(ProtoQueryResponseFactoryTest, CreateTransactionsResponse) {
   });
 }
 
+/**
+ * Checks createTransactionsPageResponse method of QueryResponseFactory
+ * @given collection of transactions, next tx hash and transactions number
+ * @when creating transactions page response via factory
+ * @then that response is created @and is well-formed
+ */
+TEST_F(ProtoQueryResponseFactoryTest, CreateTransactionsPageResponse) {
+  const HashType kQueryHash{"my_super_hash"};
+  const HashType kNextTxHash{"next_tx_hash"};
+
+  constexpr int kTransactionsNumber = 5;
+
+  std::vector<std::unique_ptr<shared_model::interface::Transaction>>
+      transactions, transactions_test_copy;
+  for (auto i = 0; i < kTransactionsNumber; ++i) {
+    auto tx = std::make_unique<shared_model::proto::Transaction>(
+        TestTransactionBuilder().creatorAccountId(std::to_string(i)).build());
+    auto tx_copy = std::make_unique<shared_model::proto::Transaction>(
+        TestTransactionBuilder().creatorAccountId(std::to_string(i)).build());
+    transactions.push_back(std::move(tx));
+    transactions_test_copy.push_back(std::move(tx_copy));
+  }
+  auto query_response = response_factory->createTransactionsPageResponse(
+      std::move(transactions), kNextTxHash, kTransactionsNumber, kQueryHash);
+
+  ASSERT_TRUE(query_response);
+  ASSERT_EQ(query_response->queryHash(), kQueryHash);
+  ASSERT_NO_THROW({
+    const auto &response =
+        boost::get<const shared_model::interface::TransactionsPageResponse &>(
+            query_response->get());
+
+    for (auto i = 0; i < kTransactionsNumber; ++i) {
```

Comment:
This test will pass if the response contains extra transactions since they will not be checked.
I suggest first verify that the size of transactions() is the same as kTransactionNumber